desafio_spring/src/main/java/meli/bootcamp/desafio_spring/services/PostService.java

```diff
@@ -62,18 +61,29 @@ public UserFollowingPostsDTO getFollowerPosts(Long userId) throws ResourceNotFou
         return allSellerFollowedPosts;
     }
 
-    public PostDTO createPost(CreatePostDTO createPost) throws ResourceNotFoundException {
+    public PostDTO createPost(CreatePostDTO createPost) throws ResourceNotFoundException, InvalidParameterException {
         Seller seller = this.userService.getSellerById(createPost.getUserId());
         Product product = this.productService.getProductById(createPost.getProductId());
-        Post newPost = new Post(createPost.getPrice(), seller, product);
-
-        this.postRepository.save(newPost);
-        
-        return PostDTO.toDTO(newPost);
+        return this.initPost(createPost.getPrice(), seller, product, null);
     }
 
     public PromotionalCountDTO getPromotionalCountBySellerId(Long userId) {
         return userService.getPromoProductsCount(userId);
     }
 
+    public PostDTO createPost(CreatePromotionalPostDTO createPromotionalPost) throws ResourceNotFoundException,
+            InvalidParameterException {
+        Seller seller = this.userService.getSellerById(createPromotionalPost.getUserId());
+        Product product = this.productService.getProductById(createPromotionalPost.getProductId());
+        return this.initPost(createPromotionalPost.getPrice(), seller, product, createPromotionalPost.getPromotion());
+    }
+
+    private PostDTO initPost(BigDecimal price, Seller seller, Product product, CreatePromotionDTO createPromotion)
```

Comment:
Acho que esse método poderia ter menos obrigações e retornar um Post, para que ele seja salvo no repositório no método do service invocado pelo controller. Talvez a palavra init poderia ser alterada para "createPost, buildPost" ou algo assim.

Reply:
Super válido! Eu estava em dúvida onde seria melhor para se salvar as alterações e decide que faria mais sentido evitar duplicação de código, mas concordo que a estrutura se torna mais coesa quando o `createPost` é quem é responsabilizado de criar um DTO.

Sobre o nome do método,  `buildPost` it is.